Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -184,6 +184,12 @@ void Init()
   StartHotplugThread();
 }
 
+void GetDevices()
```

Comment:
IMO, GetDevices is not the correct name for what this does.

The name implies that devices will be returned in some way, and without real side effects, but this is not what it does. It directly adds device to the controller interface, and here, it even shuts down a backend and reinits it.

Also, you probably don't need to shut down completely, i.e. it is unnecessary to shut down the hotplug thread and start it again simply for re-adding devices. I suggest leaving the hotplug stuff in Init and Shutdown, and moving the device enumerating part to this function.